File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        ret = list()
        for i in range(0,len(nums)-1):  
            sec = target-nums[i]          
            for j in range(i+1, len(nums)):
                if(nums[j] == sec):
                    ret.append(i)
                    ret.append(j)
        return ret
        

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.cs
public class Solution {
    public int[] TwoSum(int[] numbers, int target) {
        // Its 1 index based arr not 0
        // a+b = target
        // l+r > target => move right towards left
        // l+r <target  => move left towards right

        int left =0;
        int right=numbers.Length-1;

        while (left<right){
            while(left<right && (numbers[left] + numbers[right] > target)){
                right--;
            }
            while(left<right && (numbers[left] + numbers[right] < target)){
                left++;
            }
            if(numbers[left] + numbers[right] == target){
                return [left+1, right +1];
            }
        }
        return [];
    }
}
